[ML] Removes duplicated service code used in Single Metric Viewer

alvarezmelissa87 opened the pull request:
## Summary

Related meta issue: https://github.com/elastic/kibana/issues/176651

This PR removes duplicates created in the initial PR for adding SMV as an embeddable in dashboards:

- `ml/public/application/services/forecast_service_provider.ts` replaces `ml/public/application/services/forecast_service.js`

- `ml/public/application/timeseriesexplorer/timeseriesexplorer_utils/time_series_search_service.ts` replaces `ml/public/application/timeseriesexplorer/timeseries_search_service.ts`

- removes `ml/public/application/timeseriesexplorer/timeseriesexplorer_utils/timeseriesexplorer_utils.js`


### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

alvarezmelissa87 commented on x-pack/plugins/ml/public/application/timeseriesexplorer/timeseriesexplorer_utils/timeseriesexplorer_utils.js:
```diff
@@ -37,218 +29,6 @@ export function createTimeSeriesJobData(jobs) {
-}
-
 // Finds the chart point which is closest in time to the specified time.
 export function findNearestChartPointToTime(chartData, time) {
```
We can remove this file if this last function `findNearestChartPointToTime` isn't needed - I couldn't find it being used anywhere but wanted to get a second opinion before just removing it. 
cc @walterra, @jgowdyelastic

peteharverson replied:
I think this used to be for locating closest points in the source time series for adding anomaly markers when the chart agg interval > bucket span. But as you say, I can't see it being used anywhere now.

alvarezmelissa87 replied:
Removed in e4d02f9bf8684351d1b6c99eac225b9557cb2c96

walterra commented on x-pack/plugins/ml/public/application/timeseriesexplorer/components/forecasting_modal/forecasting_modal.js:
```diff
@@ -18,7 +18,7 @@ import { EuiButton, EuiToolTip } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import { FormattedMessage } from '@kbn/i18n-react';
-import { withKibana } from '@kbn/kibana-react-plugin/public';
+import { withKibana, context } from '@kbn/kibana-react-plugin/public';
```
This component now uses both `context` and `withKibana`, IMHO we should try to stick to one. To get rid of `withKibana()` you need to update this line in the file (not part of the diff here):

```
// old withKibana:
      const { timefilter } = this.props.kibana.services.data.query.timefilter;
// update with context:
      const { timefilter } = this.context.services.data.query.timefilter;
```

Then you should be able to remove the last line in the file (`export const ForecastingModal = withKibana(ForecastingModalUI);`) and export this component as `ForecastingModal` directly.

alvarezmelissa87 replied:
Updated in e33fb0d49ce37dcabbd55273fdefa6c425110d67

walterra reviewed (commented):
Overall nice reduction of duplicated code! 🥳 Just added some naming suggestions and one about `withKibana/context`. To play it safe, maybe let's do a PR first with functional tests of the Single Metric Viewer embedding and then get this one in afterwards?

peteharverson reviewed (commented):
Overall looks good. Manual testing of the chart inside ML and Dashboard as an embeddable all working as expected.

alvarezmelissa87 commented:
> Overall nice reduction of duplicated code! 🥳 Just added some naming suggestions and one about `withKibana/context`. To play it safe, maybe let's do a PR first with functional tests of the Single Metric Viewer embedding and then get this one in afterwards?

Thanks for taking a look @walterra - added functional tests here: https://github.com/elastic/kibana/pull/178768

alvarezmelissa87 commented on x-pack/plugins/ml/public/application/services/forecast_service_provider.ts:
```diff
@@ -23,196 +25,88 @@ export interface AggType {
 // `x-pack/plugins/ml/public/application/services/forecast_service.js` and
 // `x-pack/plugins/ml/public/application/services/forecast_service.d.ts`.
 export function forecastServiceProvider(mlApiServices: MlApiServices) {
```
Updated in e33fb0d49ce37dcabbd55273fdefa6c425110d67

alvarezmelissa87 commented on x-pack/plugins/ml/public/application/services/forecast_service_provider.ts:
```diff
@@ -23,196 +25,88 @@ export interface AggType {
 // `x-pack/plugins/ml/public/application/services/forecast_service.js` and
 // `x-pack/plugins/ml/public/application/services/forecast_service.d.ts`.
```
Removed in e33fb0d49ce37dcabbd55273fdefa6c425110d67

alvarezmelissa87 commented:
Updated with suggested changes. Also removed `timeseriesexplorer_utils.js` and removed the need for the final function in it `createTimeSeriesJobData`.

Ready for another test/look when you get a chance cc @peteharverson, @walterra 
I will hold off on merging until the functional test [PR](https://github.com/elastic/kibana/pull/178768) is merged and this PR is updated with latest from main. 👍

alvarezmelissa87 commented:
@elasticmachine merge upstream

elasticmachine commented:
Pinging @elastic/ml-ui (:ml)

peteharverson reviewed (approved):
Latest changes LGTM

jgowdyelastic commented on x-pack/plugins/ml/public/application/timeseriesexplorer/components/forecasting_modal/forecasting_modal.js:
```diff
@@ -546,4 +557,4 @@ export class ForecastingModalUI extends Component {
 }
 
-export const ForecastingModal = withKibana(ForecastingModalUI);
+export const ForecastingModal = ForecastingModalUI;
```
`ForecastingModalUI` isn't needed anymore now. The class can just called `ForecastingModal`

alvarezmelissa87 replied:
Updated in 70a4278a495651f8acb2d17f57cf79570ac71c26

jgowdyelastic commented on x-pack/plugins/ml/public/application/jobs/jobs_list/components/job_details/forecasts_table/forecasts_table.js:
```diff
@@ -21,12 +21,12 @@ import {
 import { i18n } from '@kbn/i18n';
 import { FormattedMessage } from '@kbn/i18n-react';
-import { withKibana } from '@kbn/kibana-react-plugin/public';
+import { withKibana, context } from '@kbn/kibana-react-plugin/public';
```
Looks like the same can be done here, we can remove `withKibana` as we now use `context`.

alvarezmelissa87 replied:
Updated in 70a4278a495651f8acb2d17f57cf79570ac71c26

jgowdyelastic commented on x-pack/plugins/ml/public/application/util/time_series_explorer_service.ts:
```diff
@@ -25,22 +25,39 @@ import { timeBucketsServiceFactory } from './time_buckets_service';
+  expression: string;
+}
+
+export interface ChartDataPoint {
```
Nit, these don't need to be exported anymore as they are only used inside this file.

alvarezmelissa87 replied:
Updated in 70a4278a495651f8acb2d17f57cf79570ac71c26

jgowdyelastic reviewed (approved):
Added some small comments, but on the whole LGTM.

alvarezmelissa87 commented:
@elasticmachine merge upstream

alvarezmelissa87 commented:
/ci

alvarezmelissa87 commented:
@elasticmachine merge upstream
